### PR TITLE
replace contract_call function and ContractCall* structs with unified syntax correspondents

### PR DESCRIPTION
**Pull request by mihaicalinluca**

### CODE REVIEW

File: contracts/examples/seed-nft-minter/src/distribution_module.rs [hunk 1]
```diff
@@ -33,10 +33,11 @@ pub trait DistributionModule {
...
                 continue;
             }
-            self.send()
-                .contract_call::<IgnoreValue>(distribution.address, distribution.endpoint)
-                .with_egld_or_single_esdt_transfer((token_id.clone(), token_nonce, payment_amount))
-                .with_gas_limit(distribution.gas_limit)
+            self.tx()
+                .to(&distribution.address)
+                .raw_call(distribution.endpoint)
+                .egld_or_single_esdt((token_id.clone(), token_nonce, payment_amount))
```
Comment (andrei-marinica) [inline contracts/examples/seed-nft-minter/src/distribution_module.rs:39]: should work with references 
&token_id
&payment_amount

File: contracts/feature-tests/composability/promises-features/src/call_promise_direct.rs [hunk 2]
```diff
@@ -40,13 +41,14 @@ pub trait CallPromisesDirectModule {
...
         let gas_limit = (self.blockchain().get_gas_left() - extra_gas_for_callback) * 9 / 10;
 
-        self.send()
-            .contract_call::<()>(to, endpoint_name)
-            .with_multi_token_transfer(token_payments_vec)
-            .with_gas_limit(gas_limit)
+        self.tx()
+            .to(&to)
+            .raw_call(endpoint_name)
+            .egld_or_multi_esdt(EgldOrMultiEsdtPayment::MultiEsdt(token_payments_vec))
```
Comment (andrei-marinica) [inline contracts/feature-tests/composability/promises-features/src/call_promise_direct.rs:47]: .payment(token_payments_vec)

We only use EgldOrMultiEsdtPayment if the choice is at runtime

File: contracts/feature-tests/composability/vault/src/vault.rs [hunk 3]
```diff
@@ -150,10 +150,11 @@ pub trait Vault {
...
             self.num_async_calls_sent_from_child().update(|c| *c += 1);
 
-            self.send()
-                .contract_call::<()>(caller.clone(), endpoint_name.clone())
-                .with_egld_or_single_esdt_transfer(return_payment.clone())
-                .with_gas_limit(self.blockchain().get_gas_left() / 2)
+            self.tx()
+                .to(&caller)
+                .raw_call(endpoint_name.clone())
+                .egld_or_single_esdt(return_payment.clone())
```
Comment (andrei-marinica) [inline contracts/feature-tests/composability/vault/src/vault.rs:156]: .payment(&return_payment)

no clones should be needed anymore

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

File: framework/base/src/types/interaction/tx.rs [hunk 4]
```diff
@@ -1,6 +1,7 @@
 use crate::{
     api::CallTypeApi,
     contract_base::BlockchainWrapper,
+    proxy_imports::{EgldOrEsdtTokenIdentifier, EgldOrEsdtTokenPaymentRefs, ManagedTypeApi},
```
Comment (andrei-marinica) [inline framework/base/src/types/interaction/tx.rs:4]: `proxy_imports` is just for proxies, we shouldn't use it in the framework
it's annoying that the auto-completer defaults to it
we'll purge them from the framework, but it's ok, doesn't need to be in thie PR

**Review by andrei-marinica — APPROVED** [review]

**Review by BiancaIalangi — APPROVED** [review]

**Pull request merged**